src/Instrumentation/Laravel/src/Watchers/RedisCommand/RedisCommandWatcher.php

```diff
@@ -0,0 +1,115 @@
+use Illuminate\Redis\Connections\PredisConnection;
+use Illuminate\Redis\Events\CommandExecuted;
+use OpenTelemetry\API\Instrumentation\CachedInstrumentation;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\Contrib\Instrumentation\Laravel\Watchers\Watcher;
+use OpenTelemetry\SemConv\TraceAttributes;
+use OpenTelemetry\SemConv\TraceAttributeValues;
+use RangeException;
+use RuntimeException;
+use Throwable;
+
+/**
+ * Watch the Redis Command event
+ *
+ * Call facade `Redis::enableEvents()` before using this watcher
+ */
+class RedisCommandWatcher extends Watcher
+{
+    public function __construct(
+        private CachedInstrumentation $instrumentation,
+    ) {
+    }
+
+    /** @psalm-suppress UndefinedInterfaceMethod */
+    public function register(Application $app): void
+    {
+        /** @phan-suppress-next-line PhanTypeArraySuspicious */
+        $app['events']->listen(CommandExecuted::class, [$this, 'recordRedisCommand']);
+    }
+
+    /**
+     * Record a query.
+     */
+    /** @psalm-suppress UndefinedThisPropertyFetch */
+    public function recordRedisCommand(CommandExecuted $event): void
+    {
+        $nowInNs = (int) (microtime(true) * 1E9);
+
+        $operationName = strtoupper($event->command);
+
+        /** @psalm-suppress ArgumentTypeCoercion */
+        $span = $this->instrumentation->tracer()
+            ->spanBuilder($operationName)
+            ->setSpanKind(SpanKind::KIND_CLIENT)
+            ->setStartTimestamp($this->calculateQueryStartTime($nowInNs, $event->time))
+            ->startSpan();
+
+        // See https://opentelemetry.io/docs/specs/semconv/database/redis/
+        $attributes = [
+            TraceAttributes::DB_SYSTEM => TraceAttributeValues::DB_SYSTEM_REDIS,
+            TraceAttributes::DB_NAME => $this->fetchDbIndex($event->connection),
+            TraceAttributes::DB_OPERATION => $operationName,
+            TraceAttributes::DB_QUERY_TEXT => Serializer::serializeCommand($event->command, $event->parameters),
+            TraceAttributes::SERVER_ADDRESS => $this->fetchDbHost($event->connection),
+        ];
+
+        /** @psalm-suppress PossiblyInvalidArgument */
+        $span->setAttributes($attributes);
+        $span->end($nowInNs);
+    }
+
+    private function calculateQueryStartTime(int $nowInNs, float $queryTimeMs): int
+    {
+        return (int) ($nowInNs - ($queryTimeMs * 1E6));
+    }
+
+    private function fetchDbIndex(Connection $connection): ?int
+    {
+        try {
+            if ($connection instanceof PhpRedisConnection) {
+                return $connection->client()->getDbNum();
+            } elseif ($connection instanceof PredisConnection) {
+                /** @psalm-suppress PossiblyUndefinedMethod */
+                return $connection->client()->getConnection()->getParameters()->database;
+            }
+        } catch (Throwable $e) {
+        }
+
+        return null;
+    }
+
+    private function fetchDbHost(Connection $connection): string
```

Comment:
This throws exceptions still and should probably also return null if a value cannot be resolved correctly.